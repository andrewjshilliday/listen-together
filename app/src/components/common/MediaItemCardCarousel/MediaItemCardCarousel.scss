.media-item-group-container {
  display: flex;
  width: 100%;
}

.left, .right {
  display: inline-block;
  width: 35px;

  &::before {
    content: '';
    width: 25px;
    height: 56px;
    position: relative;
    top: 30%;
    float: left;
    background-color: black;
    -webkit-mask: url(../../../assets/images/arrow.svg) no-repeat center;
    mask: url(../../../assets/images/arrow.svg) no-repeat center;
  }

  &:hover {
    cursor: pointer;
    div { background-color: red; }
  }
}

.right {
  &::before {
    float: right;
    transform: rotate(180deg);
    -ms-transform: rotate(180deg);
  }
}

.disabled {
  pointer-events: none;
  opacity: 0.1;
}

.media-item-group {
  display: inline-flex;
  flex-wrap: nowrap;
  overflow: hidden;
  scroll-behavior: smooth;
  width: calc(100% - 10px);
  margin: 0;
  padding-bottom: 15px;
}
