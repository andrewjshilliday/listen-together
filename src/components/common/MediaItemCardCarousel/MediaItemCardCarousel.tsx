import React, { useEffect, createRef } from 'react';
import { MediaItemCard } from '../../common';
import './MediaItemCardCarousel.scss';

interface MediaItemCardCarouselProps {
  items: Array<any>
}

const MediaItemCardCarousel: React.FC<MediaItemCardCarouselProps> = ({ items }) => {
  const rowRef = createRef<HTMLDivElement>();
  const leftIconRef = createRef<HTMLDivElement>();
  const rightIconRef = createRef<HTMLDivElement>();

  useEffect(() => {
    if (leftDisabled()) {
      leftIconRef.current!.classList.add('disabled');
    }
    if (rightDisabled()) {
      rightIconRef.current!.classList.add('disabled');
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [leftIconRef, rightIconRef])

  const scroll = (right: boolean) => {
    if (right) {
      rightIconRef.current!.classList.add('disabled');
      leftIconRef.current!.classList.remove('disabled');
      rowRef.current!.scrollLeft += rowRef.current!.offsetWidth;
    } else {
      leftIconRef.current!.classList.add('disabled');
      rightIconRef.current!.classList.remove('disabled');
      rowRef.current!.scrollLeft -= rowRef.current!.offsetWidth;
    }

    setTimeout(() => {
      if (right && !rightDisabled()) {
        rightIconRef.current!.classList.remove('disabled');
      } else if (!right && !leftDisabled()) {
        leftIconRef.current!.classList.remove('disabled');
      }
    }, 600);
  }

  const leftDisabled = () => {
    return rowRef.current!.scrollLeft === 0;
  }

  const rightDisabled = () => {
    return (rowRef.current!.lastChild as HTMLElement).offsetLeft < rowRef.current!.scrollLeft + rowRef.current!.offsetWidth;
  }

  return (
    <div className="media-item-group-container" id={`items-${items[0].id}-${items[items.length-1].id}`}>
      <div className="left" ref={leftIconRef} onClick={() => scroll(false)}></div>
      <div className="media-item-group" ref={rowRef}>
        {items.map((item: any) => (
          <MediaItemCard key={item.id} item={item}></MediaItemCard>
        ))}
      </div>
      <div className="right" ref={rightIconRef} onClick={() => scroll(true)}></div>
    </div>
  );
}

export default MediaItemCardCarousel;
