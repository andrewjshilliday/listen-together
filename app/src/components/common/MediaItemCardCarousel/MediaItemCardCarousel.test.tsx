import React from 'react';
import { StaticRouter } from 'react-router'
import renderer from 'react-test-renderer';
import MediaItemCardCarousel from './MediaItemCardCarousel';

jest.mock('../../../services/MusicKit');

it('renders correctly', () => {
  const tree = renderer.create(
    <StaticRouter location=".">
      <MediaItemCardCarousel items={require('../../../__mocks__/objects/Albums.mock').default} />
    </StaticRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
