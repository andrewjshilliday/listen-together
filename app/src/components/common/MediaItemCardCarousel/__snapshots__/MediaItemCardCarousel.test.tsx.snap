// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-fzoXzr RGetT"
>
  <div
    className="sc-fzpjYC bRtErT"
  >
    <div
      className="sc-fznxsB daWDfo"
      onClick={[Function]}
    />
    <div
      className="sc-fznWqX hVJbNi"
    >
      <div
        className="sc-AxjAm dYpDUQ"
      >
        <a
          href="/album/1447538299"
          onClick={[Function]}
        >
          <img
            alt="This Land"
            className="img-fluid rounded"
          />
          <span
            className="text-truncate"
          >
            This Land
          </span>
        </a>
        <span
          className="text-truncate"
        >
          Gary Clark Jr.
        </span>
      </div>
      <div
        className="sc-AxjAm dYpDUQ"
      >
        <a
          href="/album/1022109426"
          onClick={[Function]}
        >
          <img
            alt="The Story of Sonny Boy Slim"
            className="img-fluid rounded"
          />
          <span
            className="text-truncate"
          >
            The Story of Sonny Boy Slim
          </span>
        </a>
        <span
          className="text-truncate"
        >
          Gary Clark Jr.
        </span>
      </div>
      <div
        className="sc-AxjAm dYpDUQ"
      >
        <a
          href="/album/906127510"
          onClick={[Function]}
        >
          <img
            alt="Gary Clark Jr. Live"
            className="img-fluid rounded"
          />
          <span
            className="text-truncate"
          >
            Gary Clark Jr. Live
          </span>
        </a>
        <span
          className="text-truncate"
        >
          Gary Clark Jr.
        </span>
      </div>
      <div
        className="sc-AxjAm dYpDUQ"
      >
        <a
          href="/album/560639480"
          onClick={[Function]}
        >
          <img
            alt="Blak and Blu"
            className="img-fluid rounded"
          />
          <span
            className="text-truncate"
          >
            Blak and Blu
          </span>
        </a>
        <span
          className="text-truncate"
        >
          Gary Clark Jr.
        </span>
      </div>
      <div
        className="sc-AxjAm dYpDUQ"
      >
        <a
          href="/album/1200488004"
          onClick={[Function]}
        >
          <img
            alt="Live North America 2016"
            className="img-fluid rounded"
          />
          <span
            className="text-truncate"
          >
            Live North America 2016
          </span>
        </a>
        <span
          className="text-truncate"
        >
          Gary Clark Jr.
        </span>
      </div>
      <div
        className="sc-AxjAm dYpDUQ"
      >
        <a
          href="/album/560652745"
          onClick={[Function]}
        >
          <img
            alt="Blak and Blu (Deluxe Version)"
            className="img-fluid rounded"
          />
          <span
            className="text-truncate"
          >
            Blak and Blu (Deluxe Version)
          </span>
        </a>
        <span
          className="text-truncate"
        >
          Gary Clark Jr.
        </span>
      </div>
      <div
        className="sc-AxjAm dYpDUQ"
      >
        <a
          href="/album/626872587"
          onClick={[Function]}
        >
          <img
            alt="iTunes Session"
            className="img-fluid rounded"
          />
          <span
            className="text-truncate"
          >
            iTunes Session
          </span>
        </a>
        <span
          className="text-truncate"
        >
          Gary Clark Jr.
        </span>
      </div>
    </div>
    <div
      className="sc-fznxsB sc-fznJRM jgaEgn"
      onClick={[Function]}
    />
  </div>
</div>
`;
